src/components/search.js

```diff
@@ -27,16 +43,30 @@ export default class Search extends React.PureComponent {
     )
   }
 
-  setRef(c) {
-    this.input = c
-  }
-
   clear() {
+    if (this.input.value == '') return
     this.input.value = ''
+    this.search('')
+  }
+
+  handleChange() {
+    this.search(this.input.value)
+  }
+
+  handleKeyUp(e) {
+    if (e.keyCode === 13 || e.keyCode === 32) {
+      // Enter || Space
+      this.clear()
+    }
+  }
+
+  setRef(c) {
```

Comment:
Any reason why you moved this further down the file ?

Reply:
Ah, after making some changes, some things got added then deleted, and I think it just shifted existing functions. The move wasn't intended!